### ISSUE TITLE
Clean up of SvGet/SetSocketAnyType

### ISSUE THREAD

**Issue by ly29** [nortikin/sverchok#1100]
In a small effort to cleanup bit I propose that we drop this interface following #1093. 
```python
items = SvGetSocketAnyType(self, self.inputs['Item'])
#
items = self.inputs['Item'].sv_get()
```
The changes needed are small and unifying and cleaning up is good thing. Doing this cleanup will help future changes in how the code works, it is a low priority but cleanup is needed for many nodes anyway.

For bonus points set the ```deepcopy=False``` option for nodes that **don't** modify input data.

- [x] nodes/analyzer/bbox.py
- [x] nodes/analyzer/distance_pp.py
- [x] nodes/layout/wifi_out.py
- [x] nodes/vertex/math.py via #1105 
- [x] nodes/text/viewer_text_mk2.py
- [x] nodes/analyzer/kd_tree_edges_mk2.py
- [x] nodes/generator/basic_3pt_arc.py
- [x] nodes/generator/basic_spline.py
- [x] nodes/generators_extended/box_rounded.py
- [x] nodes/generators_extended/script.py
- [x] nodes/text/stethoscope.py
- [x] nodes/modifier_make/lathe.py
- [x] nodes/scene/instancer.py
- [x] nodes/viz/viewer_curves.py
- [x] nodes/viz/viewer_curves_2d.py
- [x] nodes/viz/viewer_indices.py
- [x] nodes/viz/viewer_polyline.py
- [x] nodes/generator/circle.py
- [x] nodes/generator/box.py 
- [x] nodes/generator/image.py 
- [x] nodes/analyzer/volume.py
- [x] nodes/analyzer/image_components.py
- [x] nodes/generator/random_vector_mk2.py
- [x] nodes/generators_extended/hilbert.py
- [x] nodes/generators_extended/hilbert_image.py
- [x] nodes/generators_extended/hilbert3d.py
- [x] nodes/analyzer/polygons_centers.py
- [x] nodes/analyzer/polygons_centers_mk3.py
- [x] nodes/generator/formula_shape.py
- [x] nodes/generators_extended/generative_art.py  
- [x] nodes/layout/converter.py
- [x] nodes/list_main/decompose.py
- [x] nodes/list_main/func.py
- [x] nodes/list_main/join.py
- [x] nodes/list_main/length.py
- [x] nodes/list_main/levels.py
- [x] nodes/list_main/match.py
- [x] nodes/list_main/sum_mk2.py
- [x] nodes/list_main/zip.py
- [x] nodes/list_masks/mask.py
- [x] nodes/list_masks/mask_join.py
- [x] nodes/list_struct/flip.py
- [x] nodes/list_struct/item.py
- [x] nodes/list_struct/repeater.py
- [x] nodes/list_struct/reverse.py
- [x] nodes/list_struct/shift_mk2.py
- [x] nodes/list_struct/shuffle.py
- [x] nodes/list_struct/slice.py
- [x] nodes/list_struct/split.py
- [x] nodes/list_struct/start_end.py
- [x] nodes/logic/neuro_elman.py 
- [x] nodes/logic/switch.py
- [x] nodes/modifier_make/convex_hull.py
- [x] nodes/modifier_make/join_tris.py
- [x] nodes/text/debug_print.py
- [x] nodes/vertex/normal.py
- [x] nodes/vertex/line_evaluate.py
- [x] nodes/vertex/vertices_sort.py
- [x] nodes/vertex/vector_out.py
- [x] nodes/vertex/vertices_delete_doubles.py
- [x] nodes/vertex/vector_in.py
- [x] nodes/modifier_make/uv_connect.py
- [x] nodes/modifier_make/voronoi_2d.py
- [x] nodes/modifier_make/solidify.py
- [x] nodes/number/float.py  ⚡️
- [x] nodes/matrix/euler.py
- [x] nodes/matrix/interpolation.py
- [x] nodes/matrix/shear.py
- [x] nodes/scene/text.py
- [x] nodes/text/note.py
- [x] nodes/viz/empty_out.py
- [x] nodes/modifier_change/mesh_separate.py ❄️ 
- [x] nodes/modifier_change/polygons_boom.py ❄️ 
- [x] nodes/modifier_change/polygons_to_edges.py ❄️ 
- [x] nodes/number/formula2.py  ⚡️
- [x] nodes/scene/node_remote.py   ⚡️ 
- [x] nodes/scene/obj_remote_mk2.py   ⚡️ 
- [x] nodes/vertex/drop.py  ⚡️
- [x] nodes/modifier_make/bisect.py  ⚡️
- [x] nodes/modifier_make/cross_section.py ⚡️
- [x] nodes/transforms/apply.py
- [x] nodes/transforms/mirror.py
- [x] nodes/transforms/rotation.py
- [x] nodes/transforms/move.py
- [x] nodes/transforms/scale.py
- [x] nodes/vertex/interpolation.py ❄️ 
- [x] nodes/vertex/interpolation_mk2.py ❄️ 
- [x] nodes/vertex/interpolation_stripes.py ❄️ 
- [x] nodes/modifier_make/polygons_adaptative.py ❄️ 
- [x] nodes/modifier_make/edges_adaptative.py ❄️ 
- [x] nodes/modifier_change/delete_loose.py ❄️ 
- [x] nodes/modifier_change/offset.py ⚡️
- [x] nodes/modifier_change/vertices_mask.py
- [x] nodes/modifier_change/holes_fill.py ⚡️
- [x] nodes/modifier_change/mesh_join.py  ⚡️
- [x] nodes/modifier_change/remove_doubles.py  ⚡️
- [x] nodes/modifier_make/wafel.py
- [x] nodes/modifier_make/wireframe.py ⚡️

**Comment by ly29** [comment]
I some cases this isn't even used so dropping it is just a question removing the imports.

**Comment by zeffii** [comment]
OK this needs to happen :)  

maybe tag-team this.. to ease the load

**Comment by zeffii** [comment]
i'll do mine and then reserve a few others..

**Comment by ly29** [comment]
Nice, I will continue picking them off a couple a day.

**Comment by zeffii** [comment]
also a good opportunity to do a few sanity passes on these nodes, it seems like these are the nodes that have been updated least. I won't get rid of all redundant checks immediately, but the next passes (if any) will be easier.

**Comment by zeffii** [comment]
hilbert_image is remarkably cool btw.

**Comment by zeffii** [comment]
i have to PR's ( mesh join and remove doubles ) from a while back. Those changes I do want to make permanent for those nodes.  Both nodes need to allow vert only input

**Comment by ly29** [comment]
Totally sane, add that.

**Comment by zeffii** [comment]
join_tris uses an operator that has since been changed and no longer includes a parameter called  limit, but does include two different related parameters

    bmesh.ops.join_triangles(bm, faces, cmp_seam, cmp_sharp, cmp_uvs, cmp_vcols, cmp_materials, angle_face_threshold, angle_shape_threshold)

Seen as this node never left beta I will modify it in place, it was broken anyway.

**Comment by ly29** [comment]
Hmm, the sad story behind many nodes.

**Comment by zeffii** [comment]
or https://www.blender.org/api/blender_python_api_2_78a_release/bmesh.ops.html?highlight=bmesh%20ops%20join_triangles#bmesh.ops.dissolve_limit
bmesh.ops.dissolve_limit(bm, angle_limit, use_dissolve_boundaries, verts, edges, delimit)

**Comment by ly29** [comment]
Sometimes I think that dropping a number of nodes and concentrating on a well formed set would make it more maintainable.

**Comment by zeffii** [comment]
am happy to drop join triangles in favour of a limited_dissolve node.

**Comment by ly29** [comment]
I am feeling the same about ```vertices_delete_doubles``` in favour of remove doubles. Mostly because it doesn't deal with numerical issues...
There are some considerations there, the fact remove doubles don't have any guarantee about order. 


```python
            for x in vers:
                if x not in out:
                    out.append(x)
```

**Comment by zeffii** [comment]
there will be cases where the user doesn't care about order, and other times does care. a switch to avoid the extra processing may be in order.

btw, i modified join_tris, it kind of works now... but only because i want to keep cracking on.

**Comment by ly29** [comment]
That feels like better idea than having two nodes...

**Comment by ly29** [comment]
More than halfway, yay.

Looked at some of the old nodes. Some have been in the ```old_nodes``` folder for two years. Surly they could be dropped...

**Comment by zeffii** [comment]
if only they could have a self-destruct date.

**Comment by ly29** [comment]
Yeah, anyway that is a separate issue...

**Comment by ly29** [comment]
Used ❄️ to reserve some.

**Comment by ly29** [comment]
Introducing a unique edges in polygons to edges since my code was changed a long time ago, now both options are available...

**Comment by zeffii** [comment]
i'll mark some nodes for removal with 💀

**Comment by zeffii** [comment]
yikes,, formula2 uses the update function...

**Comment by ly29** [comment]
You must mean the old one, no?

**Comment by zeffii** [comment]
no.

**Comment by zeffii** [comment]
![image](https://cloud.githubusercontent.com/assets/619340/21924460/8eb7f65a-d979-11e6-9bc7-69c5acec9645.png)

**Comment by ly29** [comment]
```python
    def update(self):
        # inputs
        multi_socket(self, min=2, start=-1, breck=True)

        if self.inputs['X'].links:
            # адаптивный сокет
            inputsocketname = 'X'
            outputsocketname = ['Result']
            changable_sockets(self, inputsocketname, outputsocketname)
```

https://github.com/nortikin/sverchok/blob/master/nodes/number/formula2.py#L66-L74

**Comment by zeffii** [comment]
i don't use that node often enough to be confident that any edits beyond the sv_get / set are non-breaking. I did a softpass  https://github.com/nortikin/sverchok/pull/1124/commits/3418d2d25de815a779c6f798e4ba3678883c578f

**Comment by ly29** [comment]
Formula is scary, many side effects with eval that makes it hard to follow...

**Comment by zeffii** [comment]
maybe that update function should be called from the strings `update=newFunction`

**Comment by ly29** [comment]
That update should be okay, adding/removing sockets is all it does.

**Comment by zeffii** [comment]
ok.. i'll move on..   do a hardpass once all svGet/Set are gone.

**Comment by zeffii** [comment]
current PR https://github.com/nortikin/sverchok/pull/1124  the end is near!

**Comment by ly29** [comment]
Seeing this kind of thin only makes me sad. 
```python
            if 'Vertices' in self.inputs and self.inputs['Vertices'].is_linked:
                Vertices = SvGetSocketAnyType(self, self.inputs['Vertices'])
            else:
                Vertices = []
            if 'X' in self.inputs and self.inputs['X'].is_linked:
                X = SvGetSocketAnyType(self, self.inputs['X'])[0]
            else:
                X = [self.x_]
            if 'Y' in self.inputs and self.inputs['Y'].is_linked:
                Y = SvGetSocketAnyType(self, self.inputs['Y'])[0]
            else:
                Y = [self.y_]
            if 'Z' in self.inputs and self.inputs['Z'].is_linked:
                Z = SvGetSocketAnyType(self, self.inputs['Z'])[0]
            else:
                Z = [self.z_]
```

**Comment by zeffii** [comment]
the interface has come a long way.

btw, is it just me or is the cross_section node a bit wonky when 'fill' is ticked.

**Comment by zeffii** [comment]
ok it's broken in master too.. the `Fill=True` routes the program code into some very iffy  `mesh.ops` territory.  a bmesh alternative is best...  this node is currently broken.

**Comment by ly29** [comment]
That code is really old an from an operator outside of Sverchok, where the constraints are different...

**Comment by zeffii** [comment]
does bisect not entirely duplicate this node's behavior?

**Comment by ly29** [comment]
No, if performs differently with filling (than the other did at least). Dealing with that is a bigger issue. At least it was.

**Comment by ly29** [comment]
Opening up some of these old nodes makes my pep8 sensitive editor cry. 😿

**Comment by zeffii** [comment]
looks like all those mesh.ops have a bmesh equivalent. i will push the svGet/Set changes and we look at fixing it later.

pylint / pep8  can be almost as immature as Drumpf.

**Comment by ly29** [comment]
So close now...

**Comment by zeffii** [comment]
would appreciate if you perused 
[mesh_join PR] (https://github.com/nortikin/sverchok/pull/1013)
[remove_doubles PR](https://github.com/nortikin/sverchok/pull/1012)

i think they can be updated without making mk2..

**Comment by ly29** [comment]
@zeffii 
Yeah I think so to.

**Comment by zeffii** [comment]
btw, stuff like this also gets on my nerves
```
   def process(.....):
        if not (self.inputs['vertices'].is_linked and self.inputs['edges'].is_linked):
            return
```

wondering if we might make a method for the custom node class

```
   def process(.....):
        if not self.is_linked_in('vertices', 'edges'):
            return
```

**Comment by ly29** [comment]
But honestly we can skip that all together.
If the inputs are are required just try to get them and trigger a exception and skip dealing with that in the node... Unless the mode of execution depends on the links existing or not.

```python
if not all(s.is_linked for s in self.inputs)
```

**Comment by ly29** [comment]
I haven't removed it always because sometimes there are other issues...

**Comment by zeffii** [comment]
in the case of `all` that's by far the simplest, but in the case of the sockets with prop_name a default/unconnected value is going to be used,  then `all` is slightly wrong.

**Comment by ly29** [comment]
If there is a sensible default that for some reason isn't exposed in a property an gotten then use the following form in ```sv_get``` 

```
socket.sv_get(default=Value)
```

For cases where a socket is  needed **do not** check for ```.is_linked``` because it is better to fail.

**Comment by ly29** [comment]
If you try to get a socket without a default value, either as property or supplied to the function a lookup error will be raised and the node shouldn't try to catch that because basically you have unfished program in the layout that doesn't make sense to execute, it would be better to catch this in a compile stage but the metadata ain't there...

**Comment by zeffii** [comment]
hey, try obj_in_mk3 -> wireframe -> vdmk2  and rotate the view .   something tells me vdmk2 is now not caching anymore..

**Comment by zeffii** [comment]
try something heavy like a monkey.

**Comment by zeffii** [comment]
or that scene update pre handler needs to ignore certain things..

**Comment by ly29** [comment]
It works well for me as before I think
(monkey + subdivision level 3)

**Comment by zeffii** [comment]
ok maybe i'll restart explorer : )

**Comment by zeffii** [comment]
monkey on subdiv 2 -> obj_in_mk3 -> vdmk2   is very slow here when rotating viewport.
 will need to resolve this in another thread :)

**Comment by ly29** [comment]
The following is smooth for me.
```python
>>> len(D.objects['Sv_0'].data.vertices)
189604
>>> len(D.objects['Sv_0'].data.polygons)
252576
```
Maybe some OpenGL changes that slow down legacy display lists? An other issue

**Comment by zeffii** [comment]
wafel not done yet.. scary node :)

**Comment by ly29** [comment]
Made soft pass there of it wafel

**Comment by ly29** [comment]
Okay made a search and basically it shows up in the following places outside of ```old_nodes```
 
- [ ] data_structure.py
- [ ] nodes/vertex/noise.py

```data_structure``` Which I am fine with, since I rather not modify ```old_nodes``` which might have any number of other issues...
And the old noise node file. Where it isn't actually used.

**Comment by ly29** [comment]
Good work, feels nice to get this done in such a large scale even if did mostly soft passes..

**Comment by zeffii** [comment]
most softpass , some have been brutal :)   ( [see hardpass](https://github.com/nortikin/sverchok/search?q=hardpass&type=Commits&utf8=%E2%9C%93) )

**Comment by zeffii** [comment]
i did test each node after each edit, this was quite a lot of work -- but definitely worth it if it allows is to progress onwards.
